### DISCUSSION TITLE
The model's RMS validation error of force, RMS training error of force did not decrease

### DISCUSSION THREAD

**Discussion by 714036303** [deepmodeling/deepmd-kit#2551] [Q&A]
I converted the data of water molecules calculated by cp2k (natom=768, nconfig=50000) into raw format with python, and then used the raw_to_set.sh command to make a data set. After the training, RMS validation error of energy, and RMS training error of energy decreased satisfactorily, but RMS validation error of force and RMS training error of force did not decrease, but fluctuated around the initial value.
![image](https://github.com/deepmodeling/deepmd-kit/assets/116813271/61d9e329-12ef-4c20-9324-e572dc8331eb)

When I train the model of water in the deepmd-kit example, everything works fine.
This is the codes when I convert the python file in raw format and training
import numpy as np
from numpy.linalg import inv

f1=open("forces.xyz", "r")
f2=open("force.raw", "w")

natom=768
nconfig=50000
convf=27.2114/0.529177

for k in range(0,nconfig):

   O=[]
   H=[]

   for i in range(0,2):
       f1.readline()

   for i in range(0,natom):
       c=f1.readline().split()
       if c[0]=="O":
          O.append(c)
       else:
          H.append(c)

   for i in range(0,len(O)):
       for j in range(1,4):
            O[i][j]=float(O[i][j])
            O[i][j]=O[i][j]*convf

   for i in range(0,len(H)):
       for j in range(1,4):
            H[i][j]=float(H[i][j])
            H[i][j]=H[i][j]*convf

   for i in range(0,len(O)):
      f2.write(str(O[i][1])+"  "+ str(O[i][2])+"   "+str(O[i][3])+ "  ")
   for i in range(0,len(H)):
      f2.write(str(H[i][1])+"  "+ str(H[i][2])+"   "+str(H[i][3])+ "  ")
   f2.write("\n")

**Comment by njzjz** [comment]
<s>`convf` should be `0.529177/27.2114` when you convert Hartree/Bohr to eV/A.</s>

**Reply by 714036303** [reply]
Thanks alot! That's the reason.

**Reply by 714036303** [reply]
> `convf` should be `0.529177/27.2114` when you convert Hartree/Bohr to eV/A.

Excuseme, I think my original convf is right. I am not understanding why convf will be the other way around?

**Reply by njzjz** [reply]
Sorry, you are right. This is not a problem.

**Reply by 714036303** [reply]
But the strange thing is that when I use the inverse to convert the units, the model's RMS validation error of force and RMS training error of force drop successfully.
![image](https://github.com/deepmodeling/deepmd-kit/assets/116813271/757084c0-26ed-4954-8c43-902cadc07ad0)

**Reply by njzjz** [reply]
What is the unit you set in the cp2k configuration?

**Reply by 714036303** [reply]
The cp2k configuration is normal, and now I am trying to increase the rcut_ smth to 5.5, then shrink the neural network to [5,10,20], and now the model can fit normally.

**Reply by 714036303** [reply]
Unfortunately, it seems that modifying the JSON file cannot solve the problem. The model is still the same as before

**Reply by njzjz** [reply]
Can you show a plot where the x-axis is the reference forces, and the y-axis is the model forces?

**Comment by njzjz** [comment]
In your script, you reorder the atoms, making O atoms ahead of H atoms. Did you do the same thing for coordinates and atomic types?

**Reply by 714036303** [reply]
yes, I did.
